src/meta/src/manager/event_log.rs

```diff
@@ -0,0 +1,295 @@
+// Copyright 2023 RisingWave Labs
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::iter;
+use std::sync::Arc;
+use std::time::{Duration, Instant, SystemTime};
+
+use etcd_client::TxnOp;
+use itertools::Itertools;
+use parking_lot::RwLock;
+use risingwave_pb::meta::event_log::EventType;
+use risingwave_pb::meta::EventLog;
+use tokio::task::JoinHandle;
+
+use crate::storage::etcd_retry_client::EtcdRetryClient;
+use crate::storage::{EtcdMetaStore, MetaStore, MetaStoreRef};
+use crate::MetaResult;
+
+pub type EventLogSender = tokio::sync::mpsc::Sender<EventLog>;
+
+const BUFFER_SIZE: usize = 1024;
+type LeaseId = i64;
+type LeaseInfo = (LeaseId, Instant);
+const CF: &[u8] = "cf/event_logs".as_bytes();
+
+fn as_etcd_client(meta_store: &MetaStoreRef) -> Option<&EtcdRetryClient> {
+    meta_store
+        .as_any()
+        .downcast_ref::<EtcdMetaStore>()
+        .map(|e| e.client())
+}
```

Comment:
Here you by-pass the `MetaStoreRef` to use the `EtcdMetaStore` directly, which looks bad considering @yezizp2012's work.